src/server/static/js/editor.js

```diff
@@ -28,17 +30,71 @@ def approval():
         Approve()
     ]))
 `;
+
+const defaultTeal = `#pragma version 6
+txn ApplicationID
+int 0
+==
+bnz main_l12
+txn OnCompletion
+int DeleteApplication
+==
+bnz main_l11
+txn OnCompletion
+int UpdateApplication
+==
+bnz main_l10
+txn OnCompletion
+int OptIn
+==
+bnz main_l9
+txn OnCompletion
+int CloseOut
+==
+bnz main_l8
+txn OnCompletion
+int NoOp
+==
+bnz main_l7
+err
+main_l7:
+int 0
+return
+main_l8:
+int 0
+return
+main_l9:
+int 0
+return
+main_l10:
+int 0
+return
+main_l11:
+int 0
+return
+main_l12:
+int 1
+return
+`
+
 const consoleMessages = [];
 
 const editorLib = {
     init() {
-        editor.session.setMode("ace/mode/python");
+        editor.getSession().setMode("ace/mode/python");
+
+        //TODO: can we define our own mode? formatting? syntax highlighting?
```

Comment:
I think this is a good idea - if we also use monaco, we might be able to port over the existing TEAL syntax highlighting rules in VSCode like the one [here](https://github.com/ObsidianLabs/vscode-algorand/blob/master/syntaxes/teal.tmLanguage.json)

Reply:
https://github.com/ajaxorg/ace/wiki/Importing-.tmtheme-and-.tmlanguage-Files-into-Ace looks like theres an example of importing that for ace

Reply:
Got this to work but punted on importing and just copy/pasta'd the resulting ace define stuff into editor.js with some tweaks to the way it gets defined.